Index: lovesey.cpp
===================================================================
--- lovesey.cpp	(revision 304)
+++ lovesey.cpp	(working copy)
@@ -665,8 +665,8 @@
          qpm = mm_gin(filename); if(qpm.isempty()) { qpm = balcar_MSq(-1,K,abs(Q),n,l); rmzeros(qpm); mm_gout(qpm,filename); }
          MSTR(K,abs(Q)); nstr[2]=109; strcpy(filename,basename); strcat(filename,nstr); strcat(filename,".mm");
          qmm = mm_gin(filename); if(qmm.isempty()) { qmm = balcar_MSq(-1,K,-abs(Q),n,l); rmzeros(qmm); mm_gout(qmm,filename); }
-         if(Q<0) { if(Q%2==0) { qpp -= qmp; qpm -= qmm; } else { qpp += qmp; qpm += qmm; } }
-         else    { if(Q%2==0) { qpp += qmp; qpm += qmm; } else { qpp -= qmp; qpm -= qmm; } }
+         if(Q<0) { if(Q%2==0) { qmp -= qpp; qmm -= qpm; } else { qmp += qpp; qmm += qpm; } }
+         else    { if(Q%2==0) { qmp += qpp; qmm += qpm; } else { qmp -= qpp; qmm -= qpm; } }
       }
       else
       {
@@ -675,8 +675,8 @@
          NSTR(K,0); nstr[2]=109; strcpy(filename,basename); strcat(filename,nstr); strcat(filename,".mm");
          qpm = mm_gin(filename); if(qpm.isempty()) { qpm = balcar_MSq(-1,K,0,n,l); rmzeros(qpm); mm_gout(qpm,filename); }
       }
-      if(density.find("sx")!=NPOS || density.find("1")!=NPOS) { if(Q<0) retval_i = (qpp-qpm)/2.;    else retval_r = (qpp-qpm)/2.; }
-      if(density.find("sy")!=NPOS || density.find("3")!=NPOS) { if(Q<0) retval_r = (qpp+qpm)/(-2.); else retval_i = (qpp+qpm)/2.; }
+      if(density.find("sx")!=NPOS || density.find("1")!=NPOS) { if(Q<0) retval_i = (qmm-qmp)/2.;    else retval_r = (qmm-qmp)/2.; }
+      if(density.find("sy")!=NPOS || density.find("3")!=NPOS) { if(Q<0) retval_r = (qmm+qmp)/(-2.); else retval_i = (qmm+qmp)/2.; }
    }
    else if(density.find("sz")!=NPOS || density.find("5")!=NPOS)
    {
@@ -687,8 +687,8 @@
          qpp = mm_gin(filename); if(qpp.isempty()) { qpp = balcar_MSq(0,K,abs(Q),n,l); rmzeros(qpp); mm_gout(qpp,filename); }
          MSTR(K,abs(Q)); nstr[2]=48; strcpy(filename,basename); strcat(filename,nstr); strcat(filename,".mm");
          qmp = mm_gin(filename); if(qmp.isempty()) { qmp = balcar_MSq(0,K,-abs(Q),n,l); rmzeros(qmp); mm_gout(qmp,filename); }
-         if(Q<0) { if(Q%2==0) qpp -= qmp; else qpp += qmp; }
-         else    { if(Q%2==0) qpp += qmp; else qpp -= qmp; }
+         if(Q<0) { if(Q%2==0) qmp -= qpp; else qmp += qpp; }
+         else    { if(Q%2==0) qmp += qpp; else qmp -= qpp; }
          retval_r = qpp/sqrt(2.);
       }
       else
@@ -710,8 +710,8 @@
          qpm = mm_gin(filename); if(qpm.isempty()) { qpm = balcar_MLq(-1,K,abs(Q),n,l); rmzeros(qpm); mm_gout(qpm,filename); }
          MSTR(K,abs(Q)); nstr[2]=109; strcpy(filename,basename); strcat(filename,nstr); strcat(filename,".mm");
          qmm = mm_gin(filename); if(qmm.isempty()) { qmm = balcar_MLq(-1,K,-abs(Q),n,l); rmzeros(qmm); mm_gout(qmm,filename); }
-         if(Q<0) { if(Q%2==0) { qpp -= qmp; qpm -= qmm; } else { qpp += qmp; qpm += qmm; } }
-         else    { if(Q%2==0) { qpp += qmp; qpm += qmm; } else { qpp -= qmp; qpm -= qmm; } }
+         if(Q<0) { if(Q%2==0) { qmp -= qpp; qmm -= qpm; } else { qmp += qpp; qmm += qpm; } }
+         else    { if(Q%2==0) { qmp += qpp; qmm += qpm; } else { qmp -= qpp; qmm -= qpm; } }
       }
       else
       {
@@ -720,8 +720,8 @@
          NSTR(K,0); nstr[2]=109; strcpy(filename,basename); strcat(filename,nstr); strcat(filename,".mm");
          qpm = mm_gin(filename); if(qpm.isempty()) { qpm = balcar_MLq(-1,K,0,n,l); rmzeros(qpm); mm_gout(qpm,filename); }
       }
-      if(density.find("sx")!=NPOS || density.find("2")!=NPOS) { if(Q<0) retval_i = (qpp-qpm)/2.;    else retval_r = (qpp-qpm)/2.; }
-      if(density.find("sy")!=NPOS || density.find("4")!=NPOS) { if(Q<0) retval_r = (qpp+qpm)/(-2.); else retval_i = (qpp+qpm)/2.; }
+      if(density.find("sx")!=NPOS || density.find("2")!=NPOS) { if(Q<0) retval_i = (qmp-qmm)/2.;    else retval_r = (qmp-qmm)/2.; }
+      if(density.find("sy")!=NPOS || density.find("4")!=NPOS) { if(Q<0) retval_r = (qmp+qmm)/(-2.); else retval_i = (qmp+qmm)/2.; }
    }
    else if(density.find("lz")!=NPOS || density.find("6")!=NPOS)
    {
@@ -732,8 +732,8 @@
          qpp = mm_gin(filename); if(qpp.isempty()) { qpp = balcar_MLq(0,K,abs(Q),n,l); rmzeros(qpp); mm_gout(qpp,filename); }
          MSTR(K,abs(Q)); nstr[2]=48; strcpy(filename,basename); strcat(filename,nstr); strcat(filename,".mm");
          qmp = mm_gin(filename); if(qmp.isempty()) { qmp = balcar_MLq(0,K,-abs(Q),n,l); rmzeros(qmp); mm_gout(qmp,filename); }
-         if(Q<0) { if(Q%2==0) qpp -= qmp; else qpp += qmp; }
-         else    { if(Q%2==0) qpp += qmp; else qpp -= qmp; }
+         if(Q<0) { if(Q%2==0) qmp -= qpp; else qmp += qpp; }
+         else    { if(Q%2==0) qmp += qpp; else qmp -= qpp; }
          retval_r = qpp/sqrt(2.);
       }
       else
